### PR TITLE
refactor: use `biome-ignore-all` for top level suppressions

### LINKED ISSUE #4305: Analyzer top-level suppressions
TL;DR

```js
/** biome-ignore lint/style/useConst: reason */

let foo = 1;
let bar = 12;
```

The top-level suppression comment will suppress `style/useConst` for the whole file

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

Closes #4305 

As requested by @arendjr , this PR uses the `biome-ignore-all` for a top-level suppression.

I also took the liberty to provide a very good diagnostic in case the `biome-ignore-all`, and because of that I had to add more information returned by the function that parses the suppression comments.

## Test Plan

Updated the snapshots, the tests and the changeset

### CODE REVIEW

File: crates/biome_analyze/src/lib.rs [hunk 1]
```diff
@@ -201,6 +203,29 @@ where
...
+                let signal = DiagnosticSignal::new(|| {
+                    let mut diagnostic = AnalyzerSuppressionDiagnostic::new(
+                        category!("suppressions/incorrect"),
+                        suppression.comment_span,
+                        "Top level suppressions can only be used at the beginning of the file.",
+                    );
+
+                    for ignore_range in &suppression.ignore_ranges {
+                        diagnostic = diagnostic.note(
+                            markup! {"Rename this to "<Emphasis>"biome-ignore"</Emphasis>}
```
Comment (arendjr) [inline crates/biome_analyze/src/lib.rs:216]:
```suggestion
                            markup! {"Rename this to "<Emphasis>"biome-ignore"</Emphasis>" or move it to the top of the file"}
```

File: crates/biome_analyze/src/lib.rs [hunk 2]
```diff
@@ -550,14 +581,24 @@ where
...
+                AnalyzerSuppressionKind::Everything => (None, None),
+                AnalyzerSuppressionKind::Rule(rule) => (Some(rule), None),
+                AnalyzerSuppressionKind::RuleInstance(rule, instance) => {
+                    (Some(rule), Some(instance))
+                }
+                AnalyzerSuppressionKind::TopLevel(rule) => (Some(rule), None),
             };
 
-            is_top_level_suppression |= kind.is_top_level();
+            // We use `text_range` because we want to check the extended range with trivia, comments in particular
```
Comment (arendjr) [inline crates/biome_analyze/src/lib.rs:593]: Do I understand correctly it is still valid to have a license comment before the suppression, for instance?

**Review by arendjr — APPROVED** [review]
Love it!

File: crates/biome_analyze/src/lib.rs (hunk 2 above)
Comment (ematipico) [reply crates/biome_analyze/src/lib.rs:593]: Yes it's still valid :) I can add more tests to cover these cases

**Pull request merged**